### PR TITLE
Remove Utf8StringLiteral conversion

### PR DESCRIPTION
**Pull request by AlekseyTs**
https://github.com/dotnet/csharplang/blob/main/meetings/2022/LDM-2022-04-18.md#target-typing-a-regular-string-literal-to-utf8-types

Test plan https://github.com/dotnet/roslyn/issues/58848

### CODE REVIEW

**Review by jcouv — APPROVED** [review]
LGTM Thanks (iteration 1)

**Comment by AlekseyTs** [comment]
@RikkiGibson, @cston, @dotnet/roslyn-compiler For the second review.

**Comment by AlekseyTs** [comment]
@RikkiGibson, @cston, https://github.com/orgs/dotnet/teams/roslyn-compiler For the second review.

**Comment by cston** [comment]
Should the [breaking change](https://github.com/dotnet/roslyn/blob/main/docs/compilers/CSharp/Compiler%20Breaking%20Changes%20-%20DotNet%207.md#utf8-string-literal-conversion) item be removed?

File: src/Compilers/CSharp/Test/Semantic/Semantics/Utf8StringsLiteralsTests.cs [hunk 1]
```diff
@@ -3402,14 +2345,10 @@ static void Main()
...
-            CompileAndVerify(comp, expectedOutput: @"array").VerifyDiagnostics();
+            CompileAndVerify(comp, expectedOutput: @"object").VerifyDiagnostics();
 
             comp = CreateCompilation(source, targetFramework: TargetFramework.NetCoreApp, options: TestOptions.DebugExe, parseOptions: TestOptions.Regular10);
-            comp.VerifyDiagnostics(
-                // (6,40): error CS8652: The feature 'UTF-8 string literals' is currently in Preview and *unsupported*. To use Preview features, use the 'preview' language version.
-                //         System.Console.WriteLine(Test(("s", 1)));
-                Diagnostic(ErrorCode.ERR_FeatureInPreview, @"""s""").WithArguments("UTF-8 string literals").WithLocation(6, 40)
-                );
+            comp.VerifyDiagnostics();
```
Comment (cston) [inline src/Compilers/CSharp/Test/Semantic/Semantics/Utf8StringsLiteralsTests.cs:2351]: >comp.VerifyDiagnostics(); [](http://example.com/codeflow?start=12&length=25)

Consider executing as well.

**Review by cston — APPROVED** [review]

**Pull request merged**